armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -800,3 +806,140 @@ def isTargetComponent(self, c):
+                    b.p.zbottom
+                )
+                self.detailedMassConservationReport["(prev) c.top"].append(b.p.ztop)
+                self.detailedMassConservationReport["(prev) c.height"].append(
+                    prevCompHeight
+                )
+        self.massConservationReport[self.a] = (
+            list(self.prevMaterialMasses.keys())
+            + [None]
+            + list(self.prevComponentMasses.keys())
+        )
+        self.massConservationReport["(prev) mass"] = (
+            list(self.prevMaterialMasses.values())
+            + [None]
+            + list(self.prevComponentMasses.values())
+        )
+
+    def populatePostExpansionLogs(self, c):
+        if self.runLogVerbosity >= self._LOGGING_VERBOSITY_THRESHOLD:
+            return
+        # populate post-expansion component mass
+        postExpMass = c.getMass()
+        self.postMaterialMasses[c.material.name] += postExpMass
+        self.postComponentMasses[c.name] += postExpMass
+        self.detailedMassConservationReport["(post) mass"].append(postExpMass)
+        self.detailedMassConservationReport["(post - prev) mass"].append(
+            postExpMass - self.preMass[c]
+        )
+        # populate post-expansion component geometry
+        try:
+            self.detailedMassConservationReport["(post) c.bottom"].append(c.zbottom)
+            self.detailedMassConservationReport["(post) c.top"].append(c.ztop)
+            self.detailedMassConservationReport["(post) c.height"].append(c.height)
+        except AttributeError:
+            # occurs when no dummy block present. The solid component do not go through the expansion changer
+            # and therefore zbotton, ztop, and height are never set. Just use the chopped parent block instead.
+            self.detailedMassConservationReport["(post) c.bottom"].append(
+                c.parent.p.zbottom
+            )
+            self.detailedMassConservationReport["(post) c.top"].append(c.parent.p.ztop)
+            self.detailedMassConservationReport["(post) c.height"].append(
+                c.parent.getHeight()
+            )
+
+    def computeMassDiffsAndPrint(self):
+        if self.runLogVerbosity >= self._LOGGING_VERBOSITY_THRESHOLD:
+            return
+        # populate remaining post-expansion criteria for conservation reports
+        self.massConservationReport["(post) mass"] = (
+            list(self.postMaterialMasses.values())
+            + [None]
+            + list(self.postComponentMasses.values())
+        )
+        # check for differences in pre and post expansion mass greater than 1e-9 grams
+        diff = [
+            round(new - orig, 9) if new else None
+            for new, orig in zip(
+                self.massConservationReport["(post) mass"],
+                self.massConservationReport["(prev) mass"],
+            )
+        ]
+        self.massConservationReport["round(post - prev, 9)"] = diff
```

Comment:
Why is this "self.massConservationReport"?  Isn't this just a local variable?  Is it used outside this method?

It appears to be a new variable that is local, but is initialized in the constructor.

Reply:
I put it on `self` to accommodate some unit testing.

Reply:
prefer returning it if its not actually needing to be an attribute.

we would have WAY more attributes if we did this every time we wanted to test.